sim/network.go

```diff
@@ -86,19 +89,21 @@ func (n *Network) Broadcast(msg *f3.GMessage) {
 					source:    msg.Sender,
 					dest:      k,
 					payload:   *msg,
-					deliverAt: n.clock + latency,
+					deliverAt: n.clock.Add(latency),
 				})
 		}
 	}
 }
 
 ///// Clock interface
 
-func (n *Network) Time() float64 {
-	return n.clock
+func (n *Network) Time() time.Time {
+	counterDuration := time.Duration(n.counter * int(time.Nanosecond))
+	n.counter++
+	return n.clock.Add(counterDuration) // To ensure not exactly equal time in simulation
```

Comment:
Please remove this hack. I think the problem is in the changes to comparisons made in the protocol code.